File: sample-infrastructure-services-domain/npm/src/commands/update-something.ts
import { inject, injectable } from "inversify";
import { ISampleUnitOfWork, ISampleUnitOfWorkFactory, TYPES as Domain } from "sample-domain";
import { IUpdateSomethingHandler, UpdateSomethingCommand } from "sample-services";
import { DomainHandlerBase } from "../domain-handler";

@injectable()
export class DomainUpdateSomethingHandler extends DomainHandlerBase<UpdateSomethingCommand, number> implements IUpdateSomethingHandler {

    readonly commandType = UpdateSomethingCommand.TYPE;

    constructor(@inject(Domain.Aggregates.ISampleUnitOfWorkFactory) uowFactory: ISampleUnitOfWorkFactory) {
        super(uowFactory);
    }

    async onHandleAsync(uow: ISampleUnitOfWork, command: UpdateSomethingCommand): Promise<number> {
        var parent = await uow.parents.getAsync(command.someId);
        if (!parent) {
            throw new Error(`Could not locate record for id (${command.someId})`);
        }
        parent.addChild(command.someNewData);
        parent.description = command.someNewData;
        var updated = await uow.parents.updateAsync(command.someId, parent);
        return updated.id;
    }

}

File: sample-infrastructure-services-domain/dotnet/src/commands/update-something.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Sample.Domain.Aggregates;
using Sample.Services.Commands;

namespace Sample.Infrastructure.Services.Domain.Commands
{
    public class DomainUpdateSomethingHandler : DomainHandlerBase<UpdateSomethingCommand, int>, IUpdateSomethingHandler
    {
        public DomainUpdateSomethingHandler(ISampleUnitOfWorkFactory uowFactory) : base(uowFactory) { }
        
        public override string CommandType => UpdateSomethingCommand.TYPE;

        public override async Task<int> OnHandleAsync(ISampleUnitOfWork uow, UpdateSomethingCommand command) 
        {
            var parent = await uow.Parents.GetAsync(command.SomeId);
            if (parent == null)
                throw new KeyNotFoundException($"Could not locate record for id ({command.SomeId})");
            parent.AddChild(command.SomeNewData);
            parent.Description = command.SomeNewData;
            var updated = await uow.Parents.UpdateAsync(command.SomeId, parent);
            return updated.Id;
        }
    }
}
